pkg/netstate/netstate.go

```diff
@@ -68,6 +83,17 @@ func (s *Server) Get(ctx context.Context, req *pb.GetRequest) (*pb.GetResponse,
 	s.logger.Debug("entering netstate get")
 
 	pointerBytes, err := s.DB.Get(req.Path)
+	
+	xAPIKeyBytes := []byte(req.XApiKey)
+	if err := validateAuth(xAPIKeyBytes); err != nil {
+		s.logger.Error("unauthorized request")
+
+		return &pb.GetResponse{
+			Pointer: []byte("Unauthorized Request"),
+		}, nil
+	}
+
+	fileValue, err := s.DB.Get(req.Path)
```

Comment:
I think another rebase leftover-- looks like line 96 can be deleted, but then the error checking on lines 97-99 need to be moved up under `pointerBytes, err := s.DB.Get(req.Path)`.

Reply:
changed!